src/Core/ClientTrait.php

```diff
@@ -176,7 +176,8 @@ private function detectProjectId(array $config)
             'httpHandler' => null,
             'projectId' => null,
             'projectIdRequired' => false,
-            'hasEmulator' => false
+            'hasEmulator' => false,
+            'useNumericProjectId' => false
```

Comment:
What do you think about calling this `preferNumericProjectId` instead?

Reply:
Sure, SGTM